### PR TITLE
Address deprecations of StaticSingleThreadedExecutor and realtime_tools/thread_priority.hpp

### PR DESCRIPTION
**Pull request by sea-bass**
### Description

We just got hit with a Rolling deprecation warning :)
https://github.com/ros2/rclcpp/pull/2598

It begs the question: We may need to cut the Jazzy release first by force-merging #3113 and #3140 before putting this in...

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/moveit/moveit2/blob/main/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://moveit.picknik.ai/humble/doc/examples/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/moveit/moveit2/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by sea-bass** [comment]
@TSNoble this is the next failure blocking your PR

**Review by sjahr — APPROVED** [review]

**Comment by sea-bass** [comment]
> Is it just the include file name changed? Does the new filename not exist in humble at all? If so, we could maybe keep humble support for a little longer by doing the "if this file exists, include it, otherwise the old file name" like we've done for the urdf.h/hpp

Yeah, good idea. I'll tack that on.

File: moveit_ros/moveit_servo/src/servo_node.cpp [hunk 1]
```diff
@@ -38,9 +38,14 @@
  *
  */
 
-#include <moveit_servo/servo_node.hpp>
+#if __has_include(<urdf/realtime_tools/realtime_helpers.hpp.hpp>)
```
Comment (mikeferguson) [inline moveit_ros/moveit_servo/src/servo_node.cpp:41]: gotta remove the urdf part!

Comment (sea-bass) [reply moveit_ros/moveit_servo/src/servo_node.cpp:41]: attention to detail is not my strength... lol

Comment (sea-bass) [reply moveit_ros/moveit_servo/src/servo_node.cpp:41]:
```suggestion
#if __has_include(<realtime_tools/realtime_helpers.hpp.hpp>)

File: moveit_ros/moveit_servo/src/servo_node.cpp [hunk 2]
```diff
@@ -38,9 +38,14 @@
  *
  */
 
-#include <moveit_servo/servo_node.hpp>
+#if __has_include(<realtime_tools/realtime_helpers.hpp.hpp>)
```
Comment (mikeferguson) [inline moveit_ros/moveit_servo/src/servo_node.cpp:41]: almost comically - you still have .hpp.hpp

Comment (sea-bass) [reply moveit_ros/moveit_servo/src/servo_node.cpp:41]: I think I need to hand back my college degree...

```suggestion
#if __has_include(<realtime_tools/realtime_helpers.hpp>)

**Comment by TSNoble** [comment]
Okay now I see why you were worried about adding warnings to our own headers 😅 

Thanks for sorting out my PR and taking a look at this 👍

**Review by mikeferguson — APPROVED** [review]

**Pull request merged**